### PR TITLE
docs: add extension policy

### PR DESCRIPTION
**Pull request by dschaller**
Signed-off-by: Derek Schaller <dschaller@lyft.com>

For an explanation of how to fill out the fields, please see the relevant section 
in [PULL_REQUESTS.md](./PULL_REQUESTS.md)

Description: Formalize the extension policy outlined in [this Google doc](https://docs.google.com/document/d/1eDQQSxqx2khTXfa2vVm4vqkyRwXYkPzZCcbjxJ2_AvA/edit#).
Risk Level: Low
Testing: N/A
Docs Changes: N/A
Release Notes: N/A

### CODE REVIEW

File: EXTENSION_POLICY.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+The following procedure will be used when proposing new extensions for inclusion in the repository:
+  1. A GitHub issue should be opened describing the proposed extension as with any major feature
+  proposal.
+  1. All extensions must be sponsored by an existing maintainer. Sponsorship means that the
+  maintainer will shepherd the extension through design/code reviews. Maintainers can self-sponsor
+  extensions if they are going to write them, shepherd them, and maintain them.
+
+  Sponsorship serves two purposes.
+    - It ensures that the extension will ultimately meet the Envoy quality bar.
```
Comment (lizan) [inline EXTENSION_POLICY.md]: add new line before this so it parsed by GitHub correctly. (same for other places below) https://github.com/envoyproxy/envoy/blob/dd05064b15aa3b7305e82b7954f142ce7ce705df/EXTENSION_POLICY.md

**Comment by junr03** [comment]
@dschaller I will let @lizan do the review for content. But I have one thought: should the content live under source/extensions?

**Comment by dschaller** [comment]
Force-pushed for the DCO check. Same content.

**Comment by dschaller** [comment]
@junr03 I did think about that but for consistency I decided to put it at the root. It seems like all the governing files live there. Happy to move it around though.

**Pull request merged**